### PR TITLE
Merge sfl branch from lp

### PR DESCRIPTION
**Pull request by mathben**
https://code.launchpad.net/openerp-canada

And add check visual improvement.

### CODE REVIEW

**Comment by bwrsandman** [comment]
PEP8 errors

```
./l10n_ca_account_check_writing/account_voucher.py:41:5: F841 local variable 'context' is assigned to but never used
./l10n_ca_account_check_writing/report/l10n_ca_check_print.py:76:121: E501 line too long (122 > 120 characters)
./l10n_ca_account_check_writing/report/l10n_ca_check_print.py:186:5: F841 local variable 'context' is assigned to but never used
```

File: l10n_ca_account_check_writing/account_voucher.py [hunk 1]
```diff
@@ -38,27 +38,38 @@ def custom_translation(s, lang):
     # should translate a string in. If we want to translate a string in another language, such as
     # a supplier's language, we have to resort to hacks such as this one. "context" is sought after
     # in the stackframe, so we have to set it.
-    context = {'lang': lang}  # NOQA
+    context = {'lang': lang}
```
Comment (bwrsandman) [inline l10n_ca_account_check_writing/account_voucher.py]: put `# noqa: _() checks local frame for context`

File: l10n_ca_account_check_writing/report/l10n_ca_check_print.py [hunk 2]
```diff
@@ -41,39 +59,137 @@ def get_all_lines(self, voucher):
...
+        journal = ac_vouch_obj.browse(cr, uid, voucher_id, context).journal_id
+        return journal.company_id.currency_id.id
+
+
+def custom_translation(s, lang):
+    # Odoo uses the current stack frame, yes, the *stack frame* to determine which language _()
+    # should translate a string in. If we want to translate a string in another language, such as
+    # a supplier's language, we have to resort to hacks such as this one. "context" is sought after
+    # in the stackframe, so we have to set it.
+    context = {'lang': lang}
```
Comment (bwrsandman) [inline l10n_ca_account_check_writing/report/l10n_ca_check_print.py]: same here

File: l10n_ca_account_check_writing/company.py [hunk 3]
```diff
@@ -25,13 +25,17 @@
...
-            #  ('bottom_ca', 'Check on bottom (CA)'),
-        ], "Choose Check layout")
+        'check_layout': fields.selection(
+            [
+                ('top', 'Check on Top'),
+                ('middle', 'Check in middle'),
+                ('bottom', 'Check on bottom'),
+                ('top_ca', 'Check on top (CA)'),
+                ('middle_ca', 'Check in middle (CA)'),
+                # ('bottom_ca', 'Check on bottom (CA)'),
```
Comment (bwrsandman) [inline l10n_ca_account_check_writing/company.py]: Take this line out please

**Comment by max3903** [comment]
:+1:

**Pull request merged**